enemies.js

```diff
@@ -238,4 +238,78 @@ class PirahnaPlant {
             }
         }
     };
-};
\ No newline at end of file
+};
+
+class FireBar {
+    constructor(game, x, y, numOfFires) {
+        Object.assign(this, { game, x, y , numOfFires});
+        let brick = new Brick(this.game, this.x, this.y, 3, "None");
+        this.game.addEntity(brick);
+
+        this.angle = 0;
+        this.fires = [];
+        for (var i = 0; i < this.numOfFires; i++) {
+            var x = i * 25 * Math.sin(this.angle);
+            var y = i * 25 * Math.cos(this.angle);
+            if(i <= 1) {
+                var fire = new FireBar_Fire(this.game, this.x + x + 12, this.y + y + 12, true);
+            } else {
+                var fire = new FireBar_Fire(this.game, this.x + x + 12, this.y + y + 12, false);
+            }
+            this.fires.push(fire);
+            this.game.addEntity(fire);
+        }
+    };
+
+    update() {
+        this.angle += this.game.clockTick;
+        if(this.angle >= 360){
+            this.angle = 0;
```

Comment:
you shouldn't set the angle to 0 on line 267 as that will make 367 degree's jump to 0 (along with every value after 360.) instead you should use mod 360 so that values outside of [0, 360) are mapped to it correctly.

the mod- operator in JS is indeed the same one as in java and python so "367 % 360" = "7"

http://www.java2s.com/Tutorial/JavaScript/0040__Operators/Modulus.htm